iDiff/cmd/iDiff.go

```diff
@@ -20,18 +21,41 @@ var iDiffCmd = &cobra.Command{
 			os.Exit(1)
 		}
 		// TODO: Use more effective mapping structure for differs
+		// TODO: Logging errors and diff results instead of just printing
 		if args[2] == "hist" {
 			diff := differs.History(args[0], args[1])
 			fmt.Println(diff)
 		} else if args[2] == "dir" {
-			diff := differs.Package(args[0], args[1])
+			diff, err := dirDiff(args[0], args[1])
+			if err != nil {
+				fmt.Println(err)
+			}
 			fmt.Println(diff)
 		} else {
 			fmt.Println("Unknown differ")
 		}
 	},
 }
 
+func dirDiff(img1, img2 string) (string, error) {
+	dir1, path1, err := utils.ImageToDir(img1)
+	if err != nil {
+		return "", err
+	}
+	dir2, path2, err := utils.ImageToDir(img2)
+	if err != nil {
+		return "", err
+	}
+	diff := differs.Package(dir1, dir2)
+
+	defer os.RemoveAll(path1)
```

Comment:
I could be wrong here, but from what I can tell, the `path`s here are paths to JSON files, and the `dir`s are the actual tar directories. if that's true, shouldn't you use `os.RemoveAll()` on the dirs (since they can contain children), and just `os.Remove()` on the paths, since they're always single files?